tests/test_localerecommender.py

```diff
@@ -60,3 +60,19 @@ def test_recommender_str(mock_s3_json_downloader):
     # Tests that the string representation of the recommender is correct
     r = LocaleRecommender()
     assert str(r) == "LocaleRecommender"
+
+
+def test_recommender_extra_data(mock_s3_json_downloader):
+    # Test that the recommender uses locale data from the "extra"
+    # section if available.
+    r = LocaleRecommender()
+    recommendations = r.recommend({}, 10, extra_data={"locale": "en"})
```

Comment:
Please add test coverage for when the client data and the extra data provided different locale values.

Reply:
Done